src/librustc/ty/error.rs

```diff
@@ -144,8 +144,7 @@ impl<'tcx> fmt::Display for TypeError<'tcx> {
                            found bound lifetime parameter {}", br)
             }
             Sorts(values) => ty::tls::with(|tcx| {
-                report_maybe_different(f, values.expected.sort_string(tcx),
-                                       values.found.sort_string(tcx))
+                write!(f, "expected {}", values.expected.sort_string(tcx))
```

Comment:
The reason why there are so many test expectation changes is due to this line here. I realize that changing this here means that all affected error messages will have to be shown in the new format, and I haven't really touched the other ones aside from E0053.